File: .skills/openclaw-skills/skills/mingo-318/image-annotation-qc/scripts/billing.py
"""
SkillPay 计费模块
用于在 skill 运行前检查用户是否已支付
"""

import os
import requests
from typing import Dict, Optional

# SkillPay 配置（从环境变量读取）
BILLING_URL = "https://skillpay.me/api/v1/billing"
API_KEY = os.environ.get('SKILL_BILLING_API_KEY', '')
SKILL_ID = os.environ.get('SKILL_ID', '')
HEADERS = {"X-API-Key": API_KEY, "Content-Type": "application/json"}

# 计费配置
PRICE_PER_CALL = 1  # 1 token


def check_payment(user_id: str) -> Dict:
    """
    检查用户是否已支付
    """
    if not API_KEY or not SKILL_ID:
        return {"success": True, "balance": 999999}
    
    try:
        resp = requests.get(
            f"{BILLING_URL}/balance", 
            params={"user_id": user_id}, 
            headers=HEADERS,
            timeout=10
        )
        if resp.status_code == 200:
            balance = resp.json().get("balance", 0)
            return {"success": balance >= PRICE_PER_CALL, "balance": balance}
    except:
        pass
    
    return {"success": True, "balance": 0}


def charge_user(user_id: str) -> Dict:
    """
    扣费用户
    返回: {"ok": bool, "balance": float, "payment_url": str or None}
    """
    if not API_KEY or not SKILL_ID:
        # 未配置计费，跳过
        return {"ok": True, "balance": 999999, "payment_url": None}
    
    try:
        resp = requests.post(
            f"{BILLING_URL}/charge", 
            headers=HEADERS, 
            json={
                "user_id": user_id,
                "skill_id": SKILL_ID,
                "amount": 0  # 扣 1 token
            },
            timeout=10
        )
        data = resp.json()
        
        if data.get("success"):
            return {"ok": True, "balance": data.get("balance", 0), "payment_url": None}
        else:
            # 余额不足，获取充值链接
            payment_url = get_payment_link(user_id)
            return {"ok": False, "balance": data.get("balance", 0), "payment_url": payment_url}
    except Exception as e:
        # 网络错误，允许通过
        return {"ok": True, "balance": 0, "payment_url": None}


def get_payment_link(user_id: str, amount: float = 8) -> str:
    """获取充值链接"""
    if not API_KEY or not SKILL_ID:
        return ""
    
    try:
        resp = requests.post(
            f"{BILLING_URL}/payment-link",
            headers=HEADERS,
            json={"user_id": user_id, "amount": amount},
            timeout=10
        )
        if resp.status_code == 200:
            return resp.json().get("payment_url", "")
    except:
        pass
    
    return ""


def require_payment(user_id: str) -> Optional[str]:
    """
    检查支付并返回支付链接（如果未支付）
    返回 None 表示已支付，返回支付链接表示需要支付
    """
    result = charge_user(user_id)
    if result.get("ok"):
        return None
    return result.get("payment_url", "")

File: .skills/openclaw-skills/skills/mybusd/bsc-dev-monitor-v2/billing.js
// ═════════════════════════════════════════════════════════════
// SkillPay Billing Integration
// スキルペイ決統合 / 결제 연동 / Интеграция биллинга SkillPay

const BILLING_API_URL = 'https://skillpay.me';
const API_KEY = 'sk_f072a786149bc07fc8730b4683dc00f3e050e72441922284ca803cdee2b994b5';
const SKILL_ID = '9469b85f-f40f-4ada-ad9c-bdd9db1167cc';

// ① Check balance / 查余额 / 残高確認 / 잔액 확인 / Проверка баланса
async function checkBalance(userId) {
  const resp = await fetch(`${BILLING_API_URL}/api/v1/billing/balance?user_id=${userId}`, {
    headers: { 'X-API-Key': API_KEY }
  });
  const data = await resp.json();
  return data.balance; // USDT amount / USDT 金额
}

// ② Charge per call / 每次调用扣费 / 呼び出しごとの課金 / 호출당 과금 / Списание за вызов
async function chargeUser(userId) {
  const resp = await fetch(`${BILLING_API_URL}/api/v1/billing/charge`, {
    method: 'POST',
    headers: {
      'X-API-Key': API_KEY,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      user_id: userId,
      skill_id: SKILL_ID,
      amount: 0.001, // USDT per call / 每次调用单价
    }),
  });
  const data = await resp.json();
  if (data.success) {
    return { ok: true, balance: data.balance };
  }
  // Insufficient balance → get payment link
  // 余额不足 → 生成充值链接
  // 残高不足 → 決済リンク生成
  // 잔액 부족 → 결제 링크 생성
  // Недостаточно средств → ссылка на оплату
  return { ok: false, balance: data.balance, paymentUrl: data.payment_url };
}

// ③ Generate payment link / 生成充值链接 / 決済リンク生成 / 결제 링크 생성 / Ссылка на оплату
async function getPaymentLink(userId, amount) {
  const resp = await fetch(`${BILLING_API_URL}/api/v1/billing/payment-link`, {
    method: 'POST',
    headers: {
      'X-API-Key': API_KEY,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      user_id: userId,
      amount
    }),
  });
  const data = await resp.json();
  return data.payment_url; // BNB Chain USDT payment link / BNB链 USDT 付款链接
}

// ════════════ Usage example / 使用示例 ════════════ //
// async function handleRequest(userId) {
//   const result = await chargeUser(userId);
//   if (result.ok) {
//     // ✅ Execute your skill logic here
//     // ✅ 在这里执行你的 Skill 逻辑
//     return { success: true };
//   } else {
//     // ❌ Insufficient balance, return payment link to user
//     // ❌ 余额不足，返回充值链接给用户
//     return { success: false, paymentUrl: result.paymentUrl };
//   }
// }

module.exports = {
  checkBalance,
  chargeUser,
  getPaymentLink
};
